prover/prover.go

```diff
@@ -91,7 +88,7 @@ func GenerateProofWithoutPersistency(
 	securityParam uint8,
 	minMemoryLayer uint,
 ) (uint64, *shared.MerkleProof, error) {
-	return GenerateProof(sig, datadir, labelHashFunc, merkleHashFunc, limit, securityParam, minMemoryLayer, persist)
+	return GenerateProof(context.TODO(), datadir, labelHashFunc, merkleHashFunc, limit, securityParam, minMemoryLayer, persist)
```

Comment:
Would it be much effort to pass in a `context` from above here? I think the only place beside tests `GenerateProofWithoutPersistency` is called is in `rpcserver.go` and the method calling it already has a `context`.

Reply:
It shouldn't be much work, I can do it. I used a `TODO()` Context here to mimick previous behaviour (previously it used a dummy global `sig` which never shut down).